From: Thomas Renninger <trenn@suse.de>
Subject: Do not modify perf bias performance setting by default at boot
References: bnc#812259,bsc#959629
Patch-Mainline: never, SLES wants to go full performance if set by BIOS
Signed-off-by: Thomas Renninger <trenn@suse.de>

It came out that on certain CPUs perf bias MSR has to be set to performance,
so that the CPU enters turbo states at all.
I will re-send this patch mainline. I guess Intel has to accept something
similar now.

---
 arch/x86/kernel/cpu/intel.c |    9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

--- a/arch/x86/kernel/cpu/intel.c
+++ b/arch/x86/kernel/cpu/intel.c
@@ -483,14 +483,11 @@ static void init_intel(struct cpuinfo_x8
 		u64 epb;
 
 		rdmsrl(MSR_IA32_ENERGY_PERF_BIAS, epb);
-		if ((epb & 0xF) == ENERGY_PERF_BIAS_PERFORMANCE) {
+		if ((epb & 0xF) == ENERGY_PERF_BIAS_PERFORMANCE)
 			printk_once(KERN_WARNING "ENERGY_PERF_BIAS:"
-				" Set to 'normal', was 'performance'\n"
+				" Is set to 'performance'\n"
 				"ENERGY_PERF_BIAS: View and update with"
-				" x86_energy_perf_policy(8)\n");
-			epb = (epb & ~0xF) | ENERGY_PERF_BIAS_NORMAL;
-			wrmsrl(MSR_IA32_ENERGY_PERF_BIAS, epb);
-		}
+				" cpupower-set(8)\n");
 	}
 }
 
